src/main/java/com/mongodb/spark/sql/connector/MongoCatalog.java

```diff
@@ -418,7 +377,7 @@ private void assertInitialized() {
   private Identifier[] filterCollections(final Identifier identifier) {
     assertInitialized();
     Assertions.ensureArgument(
-        () -> identifier.namespace().length == 1, "Namespace size must equal 1");
+        () -> identifier.namespace().length == 1, () -> "Namespace size must equal 1");
```

Comment:
I actually don't know how using the Spark API to create an identifier with a larger namespace.  A user has to go out of their way to define one - so I think the error message is fine.

Reply:
We can either assume that the length may be different from 1, and then check the argument and throw `IllegalArgumentException` (note that `Assertions.ensureArgument` is not actually an assertion despite its name, it does not throw an `AssertionError` implying that if the code is incorrect), or we assume that the length cannot be anything but 1, and do not check the argument (we can still use an assertion). But we cannot assume that the length may be different from 1 when we decide to check the argument, and then assume that it cannot be anything but 1 (and, hence, we don't need to report the actual length in the error message) when we write the text of an error message.

Reply:
We don't support namespaces greater != 1 so we throw an IllegalArgument exception in that case. 

We are asserting that and throwing the appropriate exception to the user, there is nothing that requires us to assert a condition and throw an AssertionError. The assertion class is similar to the one in the Java driver.

I think some practicality and flexibility is needed for the code reviews else nothing will ever get shipped.

Reply:
@rozza, I suspect we are talking past each other in this particular case. Because the issue is not of importance, I am resolving this discussion instead of trying to get us synchronized.